### ISSUE TITLE
steam: mangohud stopped working with soldier runtime beta

### ISSUE THREAD

**Issue by Atemu** [NixOS/nixpkgs#213366]
### Steps To Reproduce
1. Put "Steam Linux Runtime - Soldier" in client_beta
2. `MANGOHUD=1 %command%` in a Vulkan/DXVK/VK3D game
3. No mangohud

### Additional context

It works in all other places; even protontricks.

DXVK HUD still works and, oddly, other VK layers such as OBS_VKCAPTURE also still work.

@smcv Do you know of anything that has gone beta in the past week or two that might be incompatible with our still somewhat hacky setup?

### Notify maintainers

@jagajaga  @jonringer 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.7-xanmod1, NixOS, 23.05 (Stoat), 23.05.git.8ee16032bbc`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.12.0`
 - nixpkgs: `/nix/var/nix/nixpkgs`
```

**Comment by smcv** [comment]
https://gitlab.steamos.cloud/steamrt/steamrt/-/wikis/Soldier-release-notes#soldier-depot-02023012389-steampipe-build-id-10384434 is new in the last couple of weeks.

Please report this to https://github.com/ValveSoftware/steam-runtime/ with full details, including the requested logs, and I'll look at it when I'm at work. I can't debug issues without logs.

If this is a regression between the default and client_beta branches of soldier, it would be great if you could provide a log for each branch so they can be compared. Please make it obvious which one is which!

Alternatively, this might be a regression triggered by a change in your packaged MangoHUD version (particularly if it has gained new dependencies since the version that worked as expected).

**Comment by Atemu** [comment]
Yeah, I'll open a bug report with actual debug info for you steamrt guys in the relevant place. I just wanted to ask real quick whether an error like this rings some bells for you; perhaps you might have heard similar reports from other repos or know of a potentially breaking change.

**Comment by UsernameRox** [comment]
Have you tried 
`mangohud %command%`

I remember to have issues on other OSs with the `MANGOHUD=1`-command too.

**Comment by Atemu** [comment]
@UsernameRox yes I did; no difference.

**Comment by UsernameRox** [comment]
Sadly I have low Nix EXP - so I can not give you any idea for that OS. But I know some typical errors from **Steam**. A popular one is: _if you change something in the Prefix, best you can do is reboot the whole system. Even when you just uncheck a box._

**Comment by Atemu** [comment]
Nix plays no major role in this; this is most likely due to the way our sandboxing works.

What do you mean by "change something in the prefix"?

**Comment by UsernameRox** [comment]
> What do you mean by "change something in the prefix"?

**Startoptions** or anything else you change in the settings that belongs to Proton/Wine. But even controller-configuration needs a restart (at least **Steam** UI).

**Comment by Atemu** [comment]
I am not sure what you are talking about. I can change Proton version, controllers and launch options, etc. without a restart.

I understand what you mean with prefix now, you mean the WINE/Proton prefixes Steam creates. Changing something about a prefix might require to shut down the game running in said prefix but not Steam.

Anyhow, I've restarted Steam countless of times while debugging this.

**Comment by UsernameRox** [comment]
Some settings works without a restart, true. I have tinkered a lot (with **Steamtinkerlaunch** and because: _Nvidia Hardware_;-) - and some settings weren't saved correctly without a restart. I needed some time to realize that - especially when I have updated the drivers. Of course I restarted the system after that - but for some changes, in **Steam**, I had to do the same (Flatpak version).

How is your experience with **gaming on Nix**? Is it worth a try (comming from **POP!_OS**/Xanmod)? In my testing area (not my main machine) the RAM-usage from **Nix** seems to be pretty low (not many background services). Nix software is up to date, so what can go wrong?

**Comment by Atemu** [comment]
@UsernameRox please create a post on https://discourse.nixos.org/; this isn't really the place to discuss gaming on NixOS in general.

**Comment by hw-lunemann** [comment]
I opened an issue about this in the steam-runtime repo: https://github.com/ValveSoftware/steam-runtime/issues/562

The offending error seems to be: `111:51:57.315493: pressure-vessel-wrap[57602]: I: Unable to load library /nix/store/dz2hdxs2h9vqn1lqc17vk6523yqssymj-mangohud-0.6.8/$LIB/mangohud/libMangoHud.so: /home/REDACTED/.local/share/Steam/steamapps/common/SteamLinuxRuntime_soldier/pressure-vessel/libexec/steam-runtime-tools-0/x86_64-linux-gnu-inspect-library: error while loading shared libraries: libelf.so.1: cannot open shared object file: No such file or directory`

**Comment by hw-lunemann** [comment]
As @Atemu found in https://github.com/ValveSoftware/steam-runtime/issues/562 , the so we need is part of `elfutils`, shall we just plug it into the FHS environment?

**Comment by Atemu** [comment]
Yup, that should do it. Perhaps we could even remove the libelf if it was only added on the assumption that it provides libelf.so.1.

**Comment by hw-lunemann** [comment]
I doubt that was the case when it was added as a mesa dependency [two years ago](https://github.com/NixOS/nixpkgs/commit/cd29b6ff97e6bf95e537ed425fce8f0b2a92a493).
I guess I can remove it and see what happens though

**Comment by smcv** [comment]
If your Mesa is linked to `libelf.so.0` then you should keep it, if your Mesa is linked to `libelf.so.1` then you probably don't need `libelf.so.0`.

These two libraries are approximately compatible at the API level (they're both based on a third-party API originating in Solaris, if I understand correctly) and in simpler cases like the way GLib uses libelf, swapping between them is as simple as a recompile (for example Debian's GLib switched from `libelf.so.0` to `libelf.so.1` in 2014).

From the pressure-vessel/soldier/Steam point of view, the `libelf.so.1` from elfutils is the one we're interested in.

**Comment by hw-lunemann** [comment]
Looks like it's [`libelf.so.0`](https://github.com/NixOS/nixpkgs/blob/3eb7e88e736b149c1d4e9b294afc31bbde42b556/pkgs/development/libraries/mesa/default.nix#L214). 
Shall I open a pull request to include `elfutils`?

**Comment by Atemu** [comment]
@smcv thanks for the insight, this is extremely valuable information!

@hw-lunemann sure!

If we need all mesa deps anyways, we should probably simply do `multiPkgs = [ ... ] ++ mesa.buildInputs ++ mesa.propagatedBuildInputs ...`.

**Comment by mainrs** [comment]
> ### Steps To Reproduce
> 
> Steps to reproduce the behavior:
> 
>     1. Put "Steam Linux Runtime - Soldier" in client_beta
> 
>     2. `MANGOHUD=1 %command%` in a Vulkan/DXVK/VK3D game
> 
>     3. No mangohud

How exactly do I enable this? Where is the client_beta attribute located?

**Comment by hw-lunemann** [comment]
This issue can be closed @Atemu !
@mainrs Library, search "soldier", right click, properties, beta "client_beta"
